### ISSUE TITLE
can't run (TF)BartForConditionalGeneration.generation on GPU, it's speed very very very slow

### ISSUE THREAD

**Issue by TheHonestBob** [huggingface/transformers#17411]
### System Info

```shell
transformers==4.19
tensorflow-gpu==2.3
torch==1.11
```

### Who can help?

@patil-suraj@patrickvonplaten, @Narsil, @gante@Rocketknight1

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

from transformers import BertTokenizer, TFBartForConditionalGeneration
tokenizer = BertTokenizer.from_pretrained("fnlp/bart-base-chinese")
model = TFBartForConditionalGeneration.from_pretrained("fnlp/bart-base-chinese", from_pt=True)
batch_data = ['北京是[MASK]的首都']*64
for i in range(20):
    batch_dict = tokenizer.batch_encode_plus(batch_data, return_token_type_ids=False, return_tensors='tf')
    result = model.generate(**batch_dict, max_length=20)
    result = tokenizer.batch_decode(result, skip_special_tokens=True)
print(result)

### Expected behavior

```shell
1 . when I run CUDA_VISIBLE_DEVICES=1 python test.py, GPU's memory is used,GPU utilization is almost zero，and generate speed is very very very slow, cpu utilization is 100%.
2. when i replace TFBartForConditionalGeneration with BartForConditionalGeneration, GPU's memory is used,
GPU utilization is almost zero，cpu utilization greater than 100%，speed is normal,but it mean that, generate is on cpu not GPU.
```

**Comment by gante** [comment]
Hey @TheHonestBob 👋  We are aware of the generate speed problems with TensorFlow, and will be releasing an update very soon. It is not a bug, but rather how Eager Execution works, sadly. Stay tuned 🤞

**Comment by TheHonestBob** [comment]
> Hey @TheHonestBob 👋 We are aware of the generate speed problems with TensorFlow, and will be releasing an update very soon. It is not a bug, but rather how Eager Execution works, sadly. Stay tuned 🤞

thanks for your reply，what can I do before update to solve it.

**Comment by gante** [comment]
My advice would be to go with the PyTorch version, if performance is a bottleneck to you and you need something working in the next ~2 weeks. If you can afford to wait ~2 weeks, then you can have a look at the guides we are writing up at the moment :)

**Comment by TheHonestBob** [comment]
> My advice would be to go with the PyTorch version, if performance is a bottleneck to you and you need something working in the next ~2 weeks. If you can afford to wait ~2 weeks, then you can have a look at the guides we are writing up at the moment :)

OK, I will continue to pay attention no it

**Comment by gante** [comment]
@TheHonestBob -- some of the functionality to speed up has been merged recently. If you try running a modified version of your script and you have a GPU, you will see it is much much faster.

```python
import tensorflow as tf
from transformers import BertTokenizer, TFBartForConditionalGeneration
tokenizer = BertTokenizer.from_pretrained("fnlp/bart-base-chinese")
model = TFBartForConditionalGeneration.from_pretrained("fnlp/bart-base-chinese", from_pt=True)
batch_data = ['北京是[MASK]的首都']*64
xla_generate = tf.function(model.generate, jit_compile=True)
for i in range(20):
    batch_dict = tokenizer.batch_encode_plus(batch_data, return_token_type_ids=False, return_tensors='tf')
    result = xla_generate(**batch_dict, max_length=20, no_repeat_ngram_size=0, num_beams=1)
    result = tokenizer.batch_decode(result, skip_special_tokens=True)
    print(result)
```

To enable bigger values of `num_beams`, which should increase the quality of the generation, [this PR](https://github.com/huggingface/transformers/pull/17857) has to be merged first :)

**Comment by gante** [comment]
@TheHonestBob The newest release fixes this issue. Check our recent blog post -- https://huggingface.co/blog/tf-xla-generate

**Comment by TheHonestBob** [comment]
> @TheHonestBob The newest release (v4.21) fixes this issue. Check our recent blog post -- https://huggingface.co/blog/tf-xla-generate

thanks a lot, I'll try it
